[Spring boot - book_store] added book entity

Andriy-Hrynevetskyi opened the pull request:
added spring boot config files
added bookService
added bookRepository

Andriy-Hrynevetskyi committed:
added spring boot config files
added book entity
added bookService and bookRepository

Andriy-Hrynevetskyi committed:
removed credentials

bhdnchui commented on src/main/java/bookstore/model/Book.java:
```diff
@@ -0,0 +1,27 @@
+import lombok.Data;
+
+@Entity
+@Data
```
it's better to define table name visibly.

bhdnchui commented on src/main/java/bookstore/repository/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,18 @@
+public class BookRepositoryImpl implements BookRepository {
+    @Override
+    public Book save(Book book) {
+        return null;
```
Add logic here.

bhdnchui commented on src/main/java/bookstore/service/BookServiceImpl.java:
```diff
@@ -0,0 +1,27 @@
+
+    @Override
+    public Book save(Book book) {
+        return null;
```
Add logic here.

bhdnchui reviewed (changes requested):
Good job! Just a few points to consider.

Andriy-Hrynevetskyi committed:
added fixes after the code review

bhdnchui commented on src/main/java/bookstore/repository/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,46 @@
+                session.close();
+            }
+        }
+        return null;
```
Why do you return null here?

bhdnchui commented on src/main/java/bookstore/repository/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,46 @@
+
+    @Override
+    public List<Book> findAll() {
+        return null;
```
Implement logic here.

bhdnchui commented on src/main/java/bookstore/service/BookServiceImpl.java:
```diff
@@ -0,0 +1,27 @@
+
+    @Override
+    public List<Book> findAll() {
+        return null;
```
Implement logic here.

bhdnchui reviewed (changes requested):
Good job!
A few more points to consider.

Andriy-Hrynevetskyi committed:
added get all method in repository layer and removed redundant return in addmethod

d1sam reviewed (approved):
well done!

Andriy-Hrynevetskyi committed:
added bookController, bookDto,bookMapper,createBookRequestDto,EntityNotFoundException,MapperConfig.
Changed bookRepository and bookService according to new functionality.
Added all needed dependencies to pom.xml

fedorovychh commented on src/main/java/bookstore/controller/BookController.java:
```diff
@@ -0,0 +1,36 @@
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/books")
```
/api/books required in the task

fedorovychh reviewed (approved):
Approved, but consider please requirement. In general - good job!

vovasalyha commented on src/main/java/bookstore/dto/CreateBookRequestDto.java:
```diff
@@ -0,0 +1,16 @@
+import lombok.Data;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
```
Is this annotation required?

Andriy-Hrynevetskyi replied:
No i've seen that in Bohdan`s video and just added it to avoid adding id to book entity from postman

vovasalyha reviewed (approved):
LGTM
